jsonlite.bash

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env bash
+set -eo pipefail; [[ $TRACE ]] && set -x
+
+readonly VERSION="0.4.2"
+export JSONLITE_PATH=${JSONLITE_PATH:="$PWD/jsonlite.data"}
+
+
+jsonlite_version() {
+  echo "jsonlite $VERSION"
+}
+
+jsonlite_usage() {
+  jsonlite_version
+  echo "Usage: jsonlite COMMAND <command-specific-options>"
+}
+
+jsonlite_help() {
+  jsonlite_usage
+  echo
+  cat<<EOF | sort | column -c2 -t -s,
+  set <json>, Writes the json document and returns a document-id
+  get <document-id>, Retrieves a json document by document-id
+  delete <document-id>, Deletes a json document by document-id
+  drop, Drops the jsonlite database
+  help, Display this help message
+  version, Display the version number
+EOF
+  echo
+  echo "  For more information, see https://github.com/nodesocket/jsonlite"
+  echo
+}
+
+jsonlite_is_valid_uuid() {
+  if [[ "$1" =~ ^[A-F0-9]{8}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{12}$ ]]; then
+    return 0
+  fi
+
+  return 1
+}
+
+jsonlite_set() {
+  local value="$1"
+  if [[ -z "$value" ]]; then
+    echo "Missing required argument json document" 1>&2
+    exit 1;
+  fi
+
+  if [[ ! -d "$JSONLITE_PATH" ]]; then
+    mkdir -p "$JSONLITE_PATH"
+  fi
+
+  # Is this portable across distros?
+  UUID=$(uuidgen | awk '{print toupper($0)}')
+
+  if command -v json_reformat > /dev/null 2>&1; then
+    echo "$value" | json_reformat > "$JSONLITE_PATH/$UUID"
+  elif command -v jq > /dev/null 2>&1; then
+    # use the not-as-fast jq library if available
+    echo "$value" | jq '.' > "$JSONLITE_PATH/$UUID"
+  else
+    # fallback to slower json.tool
+    echo "$value" | python -m json.tool > "$JSONLITE_PATH/$UUID"
+  fi
+
+  echo "$UUID";
+}
+
+jsonlite_get() {
+  local document_id="$1"
+  if [[ -z "$document_id" ]]; then
+    echo "Missing required argument document id" 1>&2
+    exit 2;
+  fi
+
+  if ! jsonlite_is_valid_uuid "$document_id"; then
+    echo "Invalid argument document id" 1>&2
+    exit 3;
+  fi
+
+  if [[ -f "$JSONLITE_PATH/$document_id" ]]; then
+    cat "$JSONLITE_PATH/$document_id"
+  fi
+}
+
+jsonlite_delete() {
+  local document_id="$1"
+  if [[ -z "$document_id" ]]; then
+    echo "Missing required argument document id" 1>&2
+    exit 2;
+  fi
+
+  if ! jsonlite_is_valid_uuid "$document_id"; then
+    echo "Invalid argument document id" 1>&2
+    exit 3;
+  fi
+
+  if [[ -f "$JSONLITE_PATH/$document_id" ]]; then
+    rm -f "$JSONLITE_PATH/$document_id"
+  fi
+}
+
+jsonlite_drop() {
+  if [[ ! -d "$JSONLITE_PATH" ]]; then
```

Comment:
In bash, `test` and `[` are built-ins, and more or less functionally equivalent. Double-brackets allow you to do stuff like `&&`, `||`, and parameter expansion. While we don't *always* do that, think of it like the Agent J's  reference from MIB:

```
Agent J: Wait, what are you doing?
Kevin Brown/K: I always do the driving.
Agent J: Oh, no...
Kevin Brown/K: I remember that.
Agent J: No, what you remember is that you used to drive that old busted jawn. See, I drive the new hotness.
[pointing at Kay]
Agent J: Old and busted.
[pointing at himself]
Agent J: New hotness.
```

Since we are assuming bash - at least, `/bin/bash` does, as does the new shebang - we can and should just standardize on the new way of doing :)